go/cmd/dolt/commands/version.go

```diff
@@ -174,6 +175,17 @@ func checkAndPrintVersionOutOfDateWarning(curVersion string, dEnv *env.DoltEnv)
 		}
 	}
 
+	// if there were new releases in the last week, the latestRelease stored might be behind the current version built
+	isEarlier, verr := isEarlierRelease(curVersion, latestRelease)
+	if verr != nil {
+		return verr
+	}
+	if isEarlier {
+		latestRelease, verr = getLatestDoltReleaseAndRecord(path, dEnv)
+		if verr != nil {
+			return verr
+		}
+	}
 	if curVersion != latestRelease {
```

Comment:
This fix is too complicated and targets the wrong thing. The bug was we were checking for equality, when we should have been checking for earlierThan. Just do that instead, the logic to re-fetch the latest release if `isEarlier` is unnecessary.

Also add a bats test for this -- write a `latestRelease` file with an older version and assert that it doesn't trigger the warning.